core/src/main/java/com/adobe/aem/commons/assetshare/content/renditions/download/async/impl/AsyncAssetRenditionsDownloadServlet.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Asset Share Commons
+ *
+ * Copyright (C) 2021 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.adobe.aem.commons.assetshare.content.renditions.download.async.impl;
+
+import com.adobe.aem.commons.assetshare.components.actions.ActionHelper;
+import com.adobe.aem.commons.assetshare.content.AssetModel;
+import com.adobe.aem.commons.assetshare.util.RequireAem;
+import com.adobe.cq.dam.download.api.*;
+import com.google.gson.JsonObject;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.adobe.aem.commons.assetshare.content.renditions.download.async.impl.NamedRenditionDownloadTargetProcessor.*;
+
+@Component(
+        service = Servlet.class,
+        property = {
+                "sling.servlet.methods=POST",
+                "sling.servlet.resourceTypes=asset-share-commons/actions/download",
+                "sling.servlet.resourceTypes=asset-share-commons/actions/share",
```

Comment:
@davidjgonzalez curious about the actions/share resource here?

Reply:
@godanny86 actions/share was added to skip the download options models. Cart is of type actions/share, in this way cart can also initiate the downloads (assuming original rendition). Added it as part of McD requirements.

Reply:
ah ok it makes more sense in that context. So the requirement thats needed is:

1. User adds assets to the cart.
2. User opens Cart modal
3. User initiates an async download that skips the download modal and defaults to downloading original renditions only...

Reply:
That's correct, with a page configuration to enable/disable it

Reply:
Yup - that is GKs nice idea!